lib/features/account/controller/account_controller.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:cinnamon_riverpod_2/features/account/controller/account_state.dart';
+import 'package:cinnamon_riverpod_2/infra/auth/service/auth_service.dart';
+import 'package:cinnamon_riverpod_2/infra/auth/service/firebase_auth_service.dart';
+import 'package:cinnamon_riverpod_2/infra/planner/repository/trip_repository.dart';
+import 'package:cinnamon_riverpod_2/infra/traveler/repository/traveler_repository.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+final AutoDisposeNotifierProvider<AccountController, AccountState>
+    accountControllerProvider =
+    NotifierProvider.autoDispose<AccountController, AccountState>(
+  () => AccountController(),
+);
+
+class AccountController extends AutoDisposeNotifier<AccountState> {
+  FirebaseAuthService get _authService => ref.read(authServiceProvider);
+
+  TravelerRepository get _travelerRepo => ref.read(travelerRepositoryProvider);
+
+  TripRepository get _tripRepo => ref.read(tripRepositoryProvider);
+
+  Future<void> toggleNotifications(bool flag) async {
+    await _travelerRepo.updateProfileData(<String, dynamic>{
+      'sendPushNotifications': flag,
```

Comment:
this should be something like updateProfileData(currentData.copyWith(sendPushNotifications:false))